### PR TITLE
[ENH] Change data loaders and writers to minimize use of "nested_univ" input type

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### Reference Issues/PRs
#110 
Note this has introduced a bug in TSFreshExtractor described in #381. It will be easier to fix this once the rest is in.
 
#### What does this implement/fix? Explain your changes.
Sorry its so big. 

This PR does quite a lot to change the way collections are loaded and written to file. It is a little over complex to maintain backward compatibility. I am a little nervous about this, because it turns out the problem loaders are called many hundreds of times (see #118). Some of this may have been done in a previous PR, but this characterises all of the changes to the datasets package

1) meta data is loaded from the header file, and there is an option to return it
2) data is always loaded into a list of 2D numpy, one for each instance. If data is equal length, this is converted to 3D numpy. Need to do list -> numpy in all cases because we do not store number of instances in header, but its not too inefficient
3) retains the ability to return nested_univ (DataFrame with each cell a series) for backward compatibility, but otherwise removes all general usage of nested_univ. 
4) Refactors data_io into data_loaders and data_writers (it was too big)
5) Removes a load_TSC_problems function, can put it back if anyone wants it (raise an issue, see #382 )
6) Removes a test in TSFresh that seemed pointless. Testing may need further tidying/improvement
7) Single problem loaders for PLAID and japanese vowels now return list of numpy
8) The option for return_X_y == False still returns nested_univ. This need fixing (see #383), but the benchmarking module makes a lot of use of this, so will wait to fix this until benchmarking is sorted out (see #116)
9) Added tests for the arff and tsv format (see #363)
10) Simplified the writers

there may be some other changes, and I'm sure there is much that can be improved, but this is a start. Note we are not supporting time stamped data with this (we never did, but there is a tag for timestamps). This would be a good thing to implement, but it is really a separate issue.

### CODE REVIEW

File: aeon/datasets/_data_generators.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    n_cases: int
+        Number of cases.
+    series_length: int
+        Length of the series.
+    n_channels: int
+        Number of dimensions.
+
+    Returns
+    -------
+    DataFrame
```
Comment (chrisholder) [inline aeon/datasets/_data_generators.py:22]: Maybe have a description on the DataFrame

File: aeon/datasets/_data_loaders.py [hunk 2]
```diff
@@ -1,69 +1,308 @@
...
+def load_from_tsfile(
+    full_file_path_and_name,
+    replace_missing_vals_with="NaN",
+    return_meta_data=False,
+):
+    """Load time series .ts file into X and (optionally) y.
+
+    Parameters
+    ----------
+    full_file_path_and_name : string
```
Comment (chrisholder) [inline aeon/datasets/_data_loaders.py:173]: Descriptions would be nice

File: aeon/datasets/_data_loaders.py [hunk 3]
```diff
@@ -1,69 +1,308 @@
...
+        although we recommend numpy3d, because some transformers do not work with
+        numpy2d. If None will load 3D numpy or list of numpy
+        There other options, see datatypes.SCITYPE_REGISTER, but these
+        will not necessarily be supported longterm.
+    local_module: default = os.path.dirname(__file__),
+    local_dirname: default = "data"
+
+    Raises
+    ------
+    Raise ValueException if the requested return type is not supported
```
Comment (chrisholder) [inline aeon/datasets/_data_loaders.py:244]: Raises
-------
ValueError
    if the requested return type is not supported

File: aeon/datasets/_data_writers.py [hunk 4]
```diff
@@ -0,0 +1,609 @@
...
+    series_length: int, default=-1
+        Indicates the series length if they are of equal length.
+    missing_values: str, default="NaN"
+        Representation for missing values.
+    comment: str or None, default=None
+        Comment text to be inserted before the header in a block.
+    fold: str or None, default=None
+        Addon at the end of the filename, i.e. _TRAIN or _TEST.
+
+    Returns
```
Comment (chrisholder) [inline aeon/datasets/_data_writers.py:166]: Probably dont need this

File: aeon/datasets/_data_writers.py [hunk 5]
```diff
@@ -0,0 +1,609 @@
...
+def _write_data_to_tsfile(
+    X,
+    path,
+    problem_name,
+    y=None,
+    missing_values="NaN",
+    comment=None,
+    suffix="",
+):
+    """
```
Comment (chrisholder) [inline aeon/datasets/_data_writers.py:277]: Should be on first line

File: aeon/datasets/_data_writers.py [hunk 6]
```diff
@@ -0,0 +1,609 @@
...
+    y: list, ndarray or None, default=None
+        The class values for each case, optional.
+    missing_values: str, default="NaN"
+        Representation for missing values.
+    comment: str or None, default=None
+        Comment text to be inserted before the header in a block.
+    suffix: str or None, default=None
+        Addon at the end of the filename, i.e. _TRAIN or _TEST.
+
+    Returns
```
Comment (chrisholder) [inline aeon/datasets/_data_writers.py:300]: Again probably dont need this

File: aeon/datasets/_data_writers.py [hunk 7]
```diff
@@ -0,0 +1,609 @@
...
+        The class values for each case, optional.
+    comment: str or None, default=None
+        Comment text to be inserted before the header in a block.
+    fold: str or None, default=None
+        Addon at the end of the filename, i.e. _TRAIN or _TEST.
+    fit_transform: bool, default=True
+        Whether to fit the transformer prior to calling transform.
+
+    Returns
+    -------
```
Comment (chrisholder) [inline aeon/datasets/_data_writers.py:513]: Again

**Review by chrisholder — DISMISSED** [review]

File: setup.cfg [hunk 8]
```diff
@@ -12,14 +12,6 @@ addopts =
     --ignore examples
     --ignore docs
     --doctest-modules
-    --durations 10
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply setup.cfg:15]: sigh, always!

File: aeon/datasets/_data_writers.py (hunk 6 above)
Comment (TonyBagnall) [reply aeon/datasets/_data_writers.py:299]: I'll leave them I think

**Review by chrisholder — APPROVED** [review]

**Pull request merged**